src/libmongoc/tests/test-mongoc-error.c

```diff
@@ -133,6 +134,41 @@ test_has_label (void)
    BSON_ASSERT (!mongoc_error_has_label (tmp_bson ("{}"), "foo"));
 }
 
+static void
+test_state_change (void)
```

Comment:
If my earlier comment about also checking error domains is relevant, this may need to be adjusted.